TestAnalyzeRollingUpdate.test: VERIFY failed in query_actor.cpp:299 (StreamQueryProcessor) during compatibility rolling update rolling_stable-26-2-1_to_prestable-26-3

## Summary

`ydb/tests/compatibility/test_compatibility.py::TestAnalyzeRollingUpdate::test` crashes a KQP node with a `VERIFY` in `ydb/library/query_actor/query_actor.cpp:299` (condition `StreamQueryProcessor`) during the rolling update step. Reproduces in all three retries of the relwithdebinfo job; currently muted by the test framework (status MUTE in every attempt).

## Environment

- **Workflow run:** [Regression-run_compatibility #26346288798](https://github.com/ydb-platform/ydb/actions/runs/26346288798)
- **Workflow file:** `.github/workflows/regression_run_compatibility.yml` (scheduled daily at 23:00)
- **Build preset:** `relwithdebinfo`
- **Target binary built from:** `main` @ [`5d7a0facd673cca59780d6b60122edeb5ebf2813`](https://github.com/ydb-platform/ydb/commit/5d7a0facd673cca59780d6b60122edeb5ebf2813)
- **Test target:** `ydb/tests/compatibility/`
- **Test framework:** `ya make -tA` (no sanitizer in this preset; this is a plain `VERIFY` abort)

### Version triple under test

The compatibility test uses **prebuilt binaries** from S3 for the init/inter versions and the freshly built `main` binary as the target. Defaults come from [`ydb/tests/library/compatibility/versions.inc`](https://github.com/ydb-platform/ydb/blob/main/ydb/tests/library/compatibility/versions.inc):

| Role | Branch | Commit at run time |
|---|---|---|
| init   | `stable-26-1-1`  | [`ab60711e6b25c4c0598e62335178691b06358b4c`](https://github.com/ydb-platform/ydb/commit/ab60711e6b25c4c0598e62335178691b06358b4c) |
| inter  | `stable-26-2-1`  | [`91ff40e6f2d53ea0c6cd1507f1617a8a57abc048`](https://github.com/ydb-platform/ydb/commit/91ff40e6f2d53ea0c6cd1507f1617a8a57abc048) |
| target | `prestable-26-3` | [`32b27a9759660d3d69977a66770638394b76e6c6`](https://github.com/ydb-platform/ydb/commit/32b27a9759660d3d69977a66770638394b76e6c6) |

## HTML reports (relwithdebinfo, all attempts)

- try_1: https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/26346288798/ya-main-x86-64/try_1/ya-test.html
- try_2: https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/26346288798/ya-main-x86-64/try_2/ya-test.html
- try_3: https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/26346288798/ya-main-x86-64/try_3/ya-test.html

## Per-try status

| Test | try_1 | try_2 | try_3 |
|---|---|---|---|
| `TestAnalyzeRollingUpdate.test` | MUTE | MUTE | MUTE |

## Error

KQP node aborts on a `VERIFY` while processing a streaming `ANALYZE` query mid-rolling-update:

```
VERIFY failed (2026-05-24T...): NKikimr::NKqp::TQueryBase::HandleStreamQueryResult
  at ydb/library/query_actor/query_actor.cpp:299
  condition: StreamQueryProcessor
  message:   StreamQueryProcessor must be set before handling stream results

  ydb/library/query_actor/query_actor.cpp:299    NKikimr::NKqp::TQueryBase::HandleStreamQueryResult
  ydb/core/sys_view/...                          (statistics / analyze actor)
  ydb/core/kqp/...                               (executer reply path)
  ...
```

The supervised `ydbd` exits with the standard `VERIFY` abort code; the test harness then restarts the node and the rolling-update procedure proceeds, but the failure is recorded.

## Notes

- Reproduces deterministically across all three retries.
- Possibly similar previously closed report: #40261 (different signature, same test family; closed, not counted).
- `area/*` label intentionally not set — the failing code lives in `ydb/library/query_actor/` and is invoked from both KQP and sys_view/statistics paths; please re-label to the owning team.

Само падение 
```
E   GRpc memory quota was set but disabled due to issues with grpc quoter, to enable it use EnableGRpcMemoryQuota option
E   VERIFY failed (2026-05-24T09:13:53.539343Z):
E   ydb/library/query_actor/query_actor.cpp:299
E   Handle(): requirement StreamQueryProcessor failed
E   0. /-S/util/system/yassert.cpp:86: InternalPanicImpl @ 0xA3356DA
E   1. /-S/util/system/yassert.cpp:55: Panic @ 0xA32F8EB
E   2. /tmp//-S/ydb/library/query_actor/query_actor.cpp:299: Handle @ 0x13FEE133
E   3. /tmp//-S/ydb/library/actors/core/actor.cpp:394: Receive @ 0xAED5CD4
E   4. /tmp//-S/ydb/library/actors/core/executor_thread.cpp:279: Execute @ 0xAF1D67D
E   5. /tmp//-S/ydb/library/actors/core/executor_thread.cpp:468: operator() @ 0xAF2125E
E   6. /tmp//-S/ydb/library/actors/core/executor_thread.cpp:520: ProcessExecutorPool @ 0xAF20E14
E   7. /tmp//-S/ydb/library/actors/core/executor_thread.cpp:546: ThreadProc @ 0xAF219E6
E   8. /-S/util/system/thread.cpp:245: ThreadProxy @ 0xA336FF7
E   9. ??:0: ?? @ 0x7FAAA4994AC2
E   10. ??:0: ?? @ 0x7FAAA4A268BFteardown failed:
```

Падает при
rolling_stable-26-2-1_to_prestable-26-3

--- a/ydb/library/query_actor/query_actor.cpp
+++ b/ydb/library/query_actor/query_actor.cpp
@@ -295,6 +295,13 @@ void TQueryBase::ReadNextStreamPart() {
 }
 
 void TQueryBase::Handle(TEvQueryBasePrivate::TEvStreamQueryResultPart::TPtr& ev) {
+    if (Finished) {
+        // Next TEvStreamQueryResultPart can be already in the mailbox (added by
+        // the StreamQueryProcessor callback) after we called Finish() to cancel the query.
+        // In this case we are not interested in processing it anymore.
+        return;
+    }
+
     Y_ABORT_UNLESS(RunningQuery);
     Y_ABORT_UNLESS(StreamQueryProcessor);
     NumberRequests++;

